src/mem3/src/mem3_rep.erl

```diff
@@ -737,7 +737,7 @@ update_locals(Target, Db, Seq) ->
         {<<"timestamp">>, list_to_binary(mem3_util:iso8601_timestamp())}
     ],
     NewBody = mem3_rpc:save_checkpoint(Node, Name, Id, Seq, NewEntry, History),
-    {ok, _} = couch_db:update_doc(Db, #doc{id = Id, body = NewBody}, []).
+    {ok, _} = couch_db:update_doc(Db, #doc{id = Id, body = NewBody}, [bypass_write_block]).
```

Comment:
Another way we handle this is to set the io_priority to the internal replicator and check based on that.

There is a precedent in:

https://github.com/apache/couchdb/blob/6fdbe48cce8b2681b10f504608d391d703d7fd46/src/fabric/src/fabric_doc_update.erl#L148-L154

We also set that for all the other operations except mem3_rep checkpoint save, I think we just omitted by mistake there

https://github.com/apache/couchdb/blob/6fdbe48cce8b2681b10f504608d391d703d7fd46/src/mem3/src/mem3_rep.erl#L721-L740

Reply:
ooh that's neater